meilisearch/index.py

```diff
@@ -391,6 +391,128 @@ def add_documents_in_batches(
+        primary_key (optional):
+            The primary-key used in index. Ignored if already set up.
+
+        Returns
+        -------
+        update:
+            Dictionary containing an update id to track the action:
+            https://docs.meilisearch.com/reference/api/updates.html#get-an-update-status
+
+        Raises
+        ------
+        MeiliSearchApiError
+            An error containing details about why MeiliSearch can't process your request. MeiliSearch error codes are described here: https://docs.meilisearch.com/errors/#meilisearch-errors
+        """
+        return self.add_documents_raw(str_documents, primary_key, 'csv')
+
+    def add_documents_ndjson(
+        self,
+        str_documents: str,
+        primary_key: Optional[str] = None,
+    ) -> Dict[str, int]:
+        """Add string documents from a NDJSON file to the index.
+
+        Parameters
+        ----------
+        str_documents:
+            String of document from a NDJSON file.
+        primary_key (optional):
+            The primary-key used in index. Ignored if already set up.
+
+        Returns
+        -------
+        update:
+            Dictionary containing an update id to track the action:
+            https://docs.meilisearch.com/reference/api/updates.html#get-an-update-status
+
+        Raises
+        ------
+        MeiliSearchApiError
+            An error containing details about why MeiliSearch can't process your request. MeiliSearch error codes are described here: https://docs.meilisearch.com/errors/#meilisearch-errors
+        """
+        return self.add_documents_raw(str_documents, primary_key, 'jsonl')
+
+    def add_documents_raw(
+        self,
+        str_documents: str,
+        primary_key: Optional[str] = None,
+        doc_type: Optional[str] = None,
+    ) -> Dict[str, int]:
+        """Add string documents to the index.
+
+        Parameters
+        ----------
+        str_documents:
+            String of document.
+        primary_key (optional):
+            The primary-key used in index. Ignored if already set up.
+        type:
+            The type of document. Type available: 'csv', 'json', 'jsonl'
+
+        Returns
+        -------
+        update:
+            Dictionary containing an update id to track the action:
+            https://docs.meilisearch.com/reference/api/updates.html#get-an-update-status
+
+        Raises
+        ------
+        MeiliSearchApiError
+            An error containing details about why MeiliSearch can't process your request. MeiliSearch error codes are described here: https://docs.meilisearch.com/errors/#meilisearch-errors
+        """
+        if primary_key is None:
+            url = f'{self.config.paths.index}/{self.uid}/{self.config.paths.document}'
+        else:
+            primary_key = urllib.parse.urlencode({'primaryKey': primary_key})
+            url = f'{self.config.paths.index}/{self.uid}/{self.config.paths.document}?{primary_key}'
```

Comment:
Can you add a method to avoid the repetition of this lines that seems to be present in multiple methods (add_documents and update_documents if I'm not wrong)